frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -791,6 +887,55 @@ export default function InteractiveWaveformModal({
     }
   };
 
+  const calculateYLimit = (y1, y2, markerX) => {
+    const x1 = (editableStartEndTimes.endTime - editableStartEndTimes.startTime) / 100;
+    const x2 =
+      editableStartEndTimes.endTime - (editableStartEndTimes.endTime - editableStartEndTimes.startTime) / 100;
+    const slope = (y2 - y1) / (x2 - x1);
+    const yIntercept = y2 - slope * x2;
+    return markerX * slope + yIntercept;
+  };
+
+  const setBothLinesToDefault = () => {
+    setValleyLineDataToDefault();
+    setPeakLineDataToDefault();
+  };
+  const setPeakLineDataToDefault = () => {
+    let newArr = [...peakY1];
+    newArr[wellIdx] = peakValleyWindows[selectedWell].minPeaks;
+    setPeakY1(newArr);
+    newArr = [...peakY2];
+    newArr[wellIdx] = peakValleyWindows[selectedWell].minPeaks;
+    setPeakY2(newArr);
+  };
+
+  const setValleyLineDataToDefault = () => {
+    let newArr = [...valleyY1];
+    newArr[wellIdx] = peakValleyWindows[selectedWell].maxValleys;
+    setValleyY1(newArr);
+    newArr = [...valleyY2];
+    newArr[wellIdx] = peakValleyWindows[selectedWell].maxValleys;
+    setValleyY2(newArr);
+  };
+
+  const setBothLinesToNew = (newPeakY1, newPeakY2, newValleyY1, newValleyY2) => {
+    let arr = [...peakY1];
+    arr[wellIdx] = newPeakY1;
+    setPeakY1(arr);
```

Comment:
Is there a reason we can make a function for these 3 lines repeated everywhere?

Reply:
I think we might need to leave these specific statements because when added a function like this
```
const assingNewArr = (data, newValue) => {
    let newArr = [...data];
    newArr[wellIdx] = newValue;
    return newArr;
  };
```
The state stops being updated. The reason a new array needs to be created and assigned is because react does not react to changes unless the length of the array changes or a completely new Array is passed.

Reply:
I posted in an earlier comment how the function would need to be written https://github.com/CuriBio/k8s_curi/pull/486#discussion_r1169462647

Reply:
I also tried 

```
function assingNewArr(data, newValue, setState) {
    let newArr = [...data];
    newArr[wellIdx] = newValue;
    setState(newArr);
  }
```
I did not update state either

Reply:
changing `setState(newArr);` to `setState([...newArr]);` should change state